packages/camera/camera_web/example/integration_test/camera_web_test.dart

```diff
@@ -561,6 +561,52 @@ void main() {
           );
         });
 
+        testWidgets('with appropriate createCameraWithSettings options',
+            (WidgetTester tester) async {
+          when(
+            () => cameraService
+                .mapResolutionPresetToSize(ResolutionPreset.ultraHigh),
+          ).thenReturn(ultraHighResolutionSize);
+
+          final int cameraId =
+              await CameraPlatform.instance.createCameraWithSettings(
+            cameraDescription,
+            const MediaSettings(
+              resolutionPreset: ResolutionPreset.ultraHigh,
+              videoBitrate: 200000,
+              audioBitrate: 32000,
+              enableAudio: true,
+            ),
+          );
+
+          expect(
+            (CameraPlatform.instance as CameraPlugin).cameras[cameraId],
+            isA<Camera>()
+                .having(
+                  (Camera camera) => camera.textureId,
+                  'textureId',
+                  cameraId,
+                )
+                .having(
+                  (Camera camera) => camera.options,
```

Comment:
I'd recommend splitting this in multiple `expect`s, so the error message is more legible. For example, if I break the `deviceId` of the second predicate I get this error message:

```
The following TestFailure was thrown running a test:
Expected: <Instance of 'Camera'> with `textureId`: <1> and
`options`: <Instance of 'CameraOptions'>
  Actual: <Instance of 'Camera'>
   Which: has `options` with value <Instance of 'CameraOptions'>
```

Which is not great (for the `CameraOptions` at least)

Reply:
splitted